### ISSUE TITLE
rust emits a warning at line 0 when compiling an empty proc-macro crate

### ISSUE THREAD

**Issue by Eijebong** [rust-lang/rust#47692]
Steps to reproduce:

- Create a new crate
- Set it as `proc-macro = true`
- Empty the src/lib.rs file
- Compile
- Profit

```
warning: unused variable: `registrar`
--> src/lib.rs:0:1
 |
 |
 = note: #[warn(unused_variables)] on by default
 = note: to avoid this warning, consider using `_registrar` instead
```

**Comment by etaoins** [comment]
This was fixed in rust-lang/rust#47655

**Comment by estebank** [comment]
Closing as per the above comment.
